libs/SalesforceSDK/src/com/salesforce/androidsdk/rest/RestResponse.java

```diff
@@ -189,6 +190,24 @@ public JSONObject asJSONObject() throws JSONException, IOException {
 		return responseAsJSONObject;
 	}
 
+	/**
+	 * JSONObject including the response body, status code and headers.
+	 *
+	 * @return JSONObject for full response
+	 * @throws JSONException
+	 * @throws IOException
+	 */
+	public JSONObject fullResponseAsJSONObject() throws JSONException, IOException {
+		if (fullResponseAsJSONObject == null) {
+			JSONObject jsonObject = new JSONObject();
+			jsonObject.put("headers", new JSONObject(getAllHeaders()));
```

Comment:
👍 
Could we have iOS and Android full json response look similar? It will help hybrid and react native apps (they won't need platform specific parsing code).

Reply:
I'm thinking standardize on this. iOS also has some extra fields on NSURLResponse like `suggestedFilename` which I'm thinking we can still add if someone wants to go deeper on the iOS side but the shared things would match
```
{
     "response": {
         "body": XXX,
         "statusCode": XXX,
         "allHeaders":XXX,
         // Other derived NSURLResponse fields
      },
     "error": XXX 
}
```

Reply:
Yep, that format LGTM.

Reply:
Once this is set, I'll update the iOS side